File: robot-client/src/main/java/com/informatica/cliente/logica/BrazoRobot.java
package com.informatica.cliente.logica;

import java.awt.Graphics;
import lombok.Getter;

/**
 *
 * @author Vamaya
 */
@Getter
public class BrazoRobot {
    
    //VARIABLES    
    private final Dedo dedoIzq;
    private final Dedo dedoDer;
    private final Mano mano;    
    private final AnteBrazo anteBrazo;    
    private final Brazo brazo; 
    private final Base base;
    
    //CONSTRUCTOR
    public BrazoRobot() {
        
        dedoIzq = BrazoRobotFabrica.crearDedo();
        dedoDer = BrazoRobotFabrica.crearDedo();
        
        mano = BrazoRobotFabrica.crearMano(dedoIzq, dedoDer);
        
        anteBrazo = BrazoRobotFabrica.crearAntebrazo(mano);
        
        brazo = BrazoRobotFabrica.crearBrazo(anteBrazo);
                
        base= BrazoRobotFabrica.crerBase();
    }
   
    
    
    public void dibujarBrazo(Graphics lapiz){       
        
        dedoIzq.getHuesoUnido().dibujar(lapiz);
        dedoDer.getHuesoUnido().dibujar(lapiz);
        
        dedoIzq.dibujar(lapiz);
        dedoDer.dibujar(lapiz);
        
        mano.dibujar(lapiz);
        
        anteBrazo.dibujar(lapiz);
        
        brazo.dibujar(lapiz);
        
        base.dibujar(lapiz);
        
        //ARTICULACIONES
        brazo.getArticulacion().dibujar(lapiz);
        
        anteBrazo.getArticulacion().dibujar(lapiz);
        
        mano.getArticulacion().dibujar(lapiz);
        
        dedoIzq.getArticulacion().dibujar(lapiz);
        dedoDer.getArticulacion().dibujar(lapiz);
        
        dedoIzq.getHuesoUnido().getArticulacion().dibujar(lapiz);
        dedoDer.getHuesoUnido().getArticulacion().dibujar(lapiz);
        
    }   
    
}

File: my-app/src/logica/BrazoRobot.ts
import { AnteBrazo } from './AnteBrazo';
import { Brazo } from './Brazo';
import { Base } from './Base';
import { Dedo } from './Dedo';
import { Mano } from './Mano';
import { BrazoRobotFabrica } from './BrazoRobotFabrica';

export class BrazoRobot {
    dedoIzq: Dedo;
    dedoDer: Dedo;
    mano: Mano;
    anteBrazo: AnteBrazo;
    brazo: Brazo;
    base: Base;

    BrazoRobotFabrica: BrazoRobotFabrica;

    constructor() {
        this.BrazoRobotFabrica = new BrazoRobotFabrica();

        this.dedoIzq = this.BrazoRobotFabrica.crearDedo();
        this.dedoDer = this.BrazoRobotFabrica.crearDedo();
        this.mano = this.BrazoRobotFabrica.crearMano(this.dedoIzq, this.dedoDer);
        this.anteBrazo = this.BrazoRobotFabrica.crearAntebrazo(this.mano);
        this.brazo = this.BrazoRobotFabrica.crearBrazo(this.anteBrazo);
        this.base = this.BrazoRobotFabrica.crerBase();
    }

    dibujarBrazo(lapiz: any){
        this.dedoIzq.huesoUnido.dibujar(lapiz);
        this.dedoDer.huesoUnido.dibujar(lapiz);
        this.dedoIzq.dibujar(lapiz);
        this.dedoDer.dibujar(lapiz);
        this.mano.dibujar(lapiz);
        this.anteBrazo.dibujar(lapiz);
        this.brazo.dibujar(lapiz);
        this.base.dibujar(lapiz);
        // Articulaciones.
        this.brazo.articulacion.dibujar(lapiz);
        this.anteBrazo.articulacion.dibujar(lapiz);
        this.mano.articulacion.dibujar(lapiz);
        this.dedoIzq.articulacion.dibujar(lapiz);
        this.dedoDer.articulacion.dibujar(lapiz);
        this.dedoIzq.huesoUnido.articulacion.dibujar(lapiz);
        this.dedoDer.huesoUnido.articulacion.dibujar(lapiz);
    }

}
